chia/data_layer/data_layer.py

```diff
@@ -71,9 +71,6 @@ async def _start(self) -> bool:
         self.subscription_lock: asyncio.Lock = asyncio.Lock()
         if self.config.get("run_server", False):
             await self.data_layer_server.start()
-        subscriptions = await self.get_subscriptions()
-        for subscription in subscriptions:
-            await self.wallet_rpc.dl_track_new(subscription.tree_id)
         self.periodically_fetch_data_task: asyncio.Task[Any] = asyncio.create_task(self.periodically_fetch_data())
```

Comment:
Do we need to do anything on shutdown if this task is still pending (ie start data but never start wallet and then stop data)

Reply:
no, it'll just stay in the first loop until shutdown (as it never received `success=True`), then ignore the second loop (as it received `self._shut_down = True`). also the task is previously cancelled in `_await_closed` on shutdown.